src/main/java/org/candlepin/resource/RhsmApiCompatResource.java

```diff
@@ -0,0 +1,141 @@
+/*
+ * Copyright (c) 2009 - 2025 Red Hat, Inc.
+ *
+ * This software is licensed to you under the GNU General Public License,
+ * version 2 (GPLv2). There is NO WARRANTY for this software, express or
+ * implied, including the implied warranties of MERCHANTABILITY or FITNESS
+ * FOR A PARTICULAR PURPOSE. You should have received a copy of GPLv2
+ * along with this software; if not, see
+ * http://www.gnu.org/licenses/old-licenses/gpl-2.0.txt.
+ *
+ * Red Hat trademarks are not licensed under GPLv2. No permission is
+ * granted to use or replicate Red Hat trademarks that are incorporated
+ * in this software or its documentation.
+ */
+package org.candlepin.resource;
+
+import org.candlepin.config.ConfigProperties;
+import org.candlepin.config.Configuration;
+import org.candlepin.dto.api.server.v1.ConsumerEntitlementCountsDTO;
+import org.candlepin.dto.api.server.v1.RhsmApiConsumerEntitlementCountsQueryDTO;
+import org.candlepin.dto.api.server.v1.RhsmApiEntitlementCountDTO;
+import org.candlepin.exceptions.BadRequestException;
+import org.candlepin.model.ConsumerEntitlementCount;
+import org.candlepin.model.RhsmApiCompatCurator;
+import org.candlepin.paging.PageRequest;
+import org.candlepin.paging.PageRequest.Order;
+import org.candlepin.paging.PagingUtilFactory;
+import org.candlepin.resource.server.v1.RhsmapiApi;
+
+import com.google.inject.Inject;
+import com.google.inject.persist.Transactional;
+
+import org.jboss.resteasy.core.ResteasyContext;
+import org.xnap.commons.i18n.I18n;
+
+import java.util.ArrayList;
+import java.util.List;
+import java.util.Objects;
+import java.util.stream.Stream;
+
+import javax.validation.Valid;
+import javax.validation.constraints.NotNull;
+
+/**
+ * Entry point for endpoints that are dedicated for RHSM API for compatibility reasons.
+ */
+public class RhsmApiCompatResource implements RhsmapiApi {
+    private final RhsmApiCompatCurator rhsmApiCompatCurator;
+    private final PagingUtilFactory pagingUtilFactory;
+    private final Configuration config;
+    private final I18n i18n;
+
+    private final int pageLimit;
+
+    @Inject
+    public RhsmApiCompatResource(RhsmApiCompatCurator rhsmApiCompatCurator,
+        PagingUtilFactory pagingUtilFactory,
+        Configuration config,
+        I18n i18n) {
+
+        this.rhsmApiCompatCurator = Objects.requireNonNull(rhsmApiCompatCurator);
+        this.pagingUtilFactory = Objects.requireNonNull(pagingUtilFactory);
+        this.config = Objects.requireNonNull(config);
+        this.i18n = Objects.requireNonNull(i18n);
+
+        this.pageLimit = this.config.getInt(ConfigProperties.RHSM_API_PAGE_LIMIT);
+    }
+
+    // POST /rhsmapi/consumers/entitlement_counts
+
+    @Override
+    @Transactional
+    public Stream<ConsumerEntitlementCountsDTO> listConsumerEntitlementCounts(
+        @Valid @NotNull RhsmApiConsumerEntitlementCountsQueryDTO rhsmApiConsumerEntitlementCountsQueryDTO,
+        Integer page, Integer perPage) {
```

Comment:
It's not necessary to explicitly define these parameters in the spec and method. Since we have to do null checks anyway, it may be better here to pull it from the req. context and only need to do special max per-page validation.

It also lets us explicitly set the sorting parameters on the shared instance instead of overwriting it as we do below.

Reply:
Updated